src/client/files.ts

```diff
@@ -99,6 +99,23 @@ const fileListToArray = (fileList: FileList) => {
   return files;
 };
 
+export async function uploadRemoteFile<T extends AllowedFileTypes>(
+  url: string,
+  allowedFileTypes: T
+): Promise<AllowedFileTypesToObject<T>> {
+  const blob = await fetch(url).then((r) => r.blob());
+  return (
+    await uploadFiles(
+      [
+        new File([blob], url.split("/").pop() ?? "remote-file", {
```

Comment:
nitpick: It might be a bit better to use `new URL(url).pathname.split('/').pop()` here, so that the query and fragment are not part of the file name (e.g. `https://example.com/foo/bar.png?something#something` -> `bar.png`)